zebra/zserv.c

```diff
@@ -842,14 +848,14 @@ static void zserv_fec_register(struct zserv *client, u_short length)
 	 * The minimum amount of data that can be sent for one fec
 	 * registration
 	 */
-	if (length < ZEBRA_MIN_FEC_LENGTH) {
+	if (hdr->length < ZEBRA_MIN_FEC_LENGTH) {
 		zlog_err(
 			"fec_register: Received a fec register of length %d, it is of insufficient size to properly decode",
-			length);
+			hdr->length);
 		return;
 	}
 
-	while (l < length) {
+	while (l < hdr->length) {
```

Comment:
as a question -> the decode function ( that I commented on in the previous commit ) does not take into account subtraction of the zapi header.  Is this going to be a problem here?

Reply:
No, the handler functions expect the stream pointer to be at the first byte after the header.
`zserv_process_messages` resets the stream pointer and then extracts the header before calling down to the handlers, so they get the offset they expect.